### ISSUE TITLE
How to return a list with conditionals expressions

### ISSUE THREAD

**Issue by ishantanu** [hashicorp/terraform#21374]
### Terraform Version

```
Terraform v0.11.11
```

### Terraform Configuration Files

```hcl
locals {
  sec_group_list = "${split(",", var.security_group_id)}"

}

${var.existing_vpc ? local.sec_group_list : aws_security_group.rancher_security_group.id}
```

### Debug Output

```
At column 3, line 1: conditional operator cannot be used with list values in:

${var.existing_vpc ? local.sec_group_list : aws_security_group.xyz_security_group.id}
```

### Expected Behavior

A list should be a valid return type in case of conditionals.

### Actual Behavior

conditional operator cannot be used with list values

### Additional Context

I have a case where I have to create a security group if the user does not provide VPC ID. The type of the `security_group` var should be a list. How can I use conditional operator to return a list instead of a string.

**Comment by apparentlymart** [comment]
Hi @ishantanu,

In Terraform 0.11, the usual workaround is to use delimited strings and then split the result:

```hcl
${split(",", var.existing_vpc ? var.security_group_id : join(",", aws_security_group.rancher_security_group.id))}
```

This has already been fixed in the master branch ready for the 0.12.0 release, which is currently in [release candidate 1](https://www.hashicorp.com/blog/releasing-terraform-0-12-0-rc1) and its final release is expected very soon. In that new version, what you tried should work exactly as you wrote it.

Since there's already a fix for this in the `master` branch, I'm going to close this out. I hope the `join`/`split` workaround above unblocks you for the moment and that you're able to upgrade to the new version soon to clean up that workaround.

**Comment by ishantanu** [comment]
@apparentlymart This is not working. It still returns string. The case is:

```
if existing_vpc == true: return list(var.security_group_id)
else: return list(aws_security_group.rancher_security_group.id)
```

I am seeing this error:
```
amazonec2_config.0.security_group: should be a list
```
